Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/fragment/SubmissionDispatcherFragment.kt

```diff
@@ -46,6 +48,32 @@ class SubmissionDispatcherFragment : Fragment(R.layout.fragment_submission_dispa
                         SubmissionDispatcherFragmentDirections
                             .actionSubmissionDispatcherFragmentToSubmissionConsentFragment()
                     )
+                is SubmissionNavigationEvents.NavigateToCreateProfile ->
+                    doNavigate(
+                        SubmissionDispatcherFragmentDirections
+                            .actionSubmissionDispatcherFragmentToRapidTestProfileNavGraph()
+                    )
+
+                is SubmissionNavigationEvents.NavigateToOpenProfile -> {
+                    val ratGraph = findNavController().graph.findNode(R.id.rapid_test_profile_nav_graph) as NavGraph
+                    ratGraph.startDestination = R.id.ratProfileQrCodeFragment
+
+                    doNavigate(
+                        SubmissionDispatcherFragmentDirections
+                            .actionSubmissionDispatcherFragmentToRapidTestProfileNavGraph()
+                    )
+                }
+            }
+        }
+
+        viewModel.profileCardId.observe(viewLifecycleOwner) { layoutId ->
+            binding.ratProfileCard.viewStub?.apply {
+                layoutResource = layoutId
```

Comment:
Why this approach over `layout.isGone=true/false`?

Reply:
Saw the code in the viewmodel now. Interesting pattern, never used that before, do you have experience with it or just something to try :grin: ?

Reply:
Yes, Intended to use that here, the cards will not be displayed together at all, So Instead of inflating them all the time even if one of them is hidden, I inflate what is needed, It kind of `lazy` way to inflate the right card